packages/cli/src/tools/__tests__/packageManager-test.ts

```diff
@@ -83,9 +100,65 @@ describe('npm', () => {
   });
 });
 
+describe('bun', () => {
+  it('should install', () => {
+    jest.spyOn(bun, 'getBunVersionIfAvailable').mockImplementation(() => true);
+    jest
+      .spyOn(bun, 'isProjectUsingBun')
+      .mockImplementation(() => './path/to/bun.lockb');
```

Comment:
We could avoid mocking implementation details by mocking `execa` and setting `mockResolvedValue` manually based on our expectations of what's to be returned from a system call. And setting fixtures including these files with e.g. `__setMockFilesystem`. There are examples of doing this in the repo, so hopefully it's not too much of an ask

Reply:
not sure if we can avoid mocking this implementation details - we need to check if it's using the proper package manager since if it's not available, the `npm` is used as fallback (same way that yarn works). This tests are checking if execa is called with proper package manager rather than it installs/uninstalls dependencies correctly

However in the `init` command we should not ever get into this kind of situation when it's using npm instead of yarn/bun because now we have a check in the beginning of the code execution, throwing an error if package manager is not available to use